nats.go

```diff
@@ -2535,13 +2544,21 @@ func (nc *Conn) PublishRequest(subj, reply string, data []byte) error {
 // Used for handrolled itoa
 const digits = "0123456789"
 
-// publish is the internal function to publish messages to a nats-server.
-// Sends a protocol data message by queuing into the bufio writer
-// and kicking the flush go routine. These writes should be protected.
 func (nc *Conn) publish(subj, reply string, data []byte) error {
 	if nc == nil {
 		return ErrInvalidConnection
 	}
+	call := publish
+	if nc.Opts.PublishWrapper != nil {
```

Comment:
I think this should be more dynamic and include more that one. Also should be able to have different ones based on subject I think as well.

Reply:
@derekcollison these are great ideas! I hadn't thought of allowing for different wrappers based on subject, that sounds awesome. I'll code up an example and share it with you.